pedalboard/plugins/PitchShift.h

```diff
@@ -23,41 +23,38 @@ namespace py = pybind11;
 #include "../RubberbandPlugin.h"
 
 namespace Pedalboard {
-class PitchShift : public RubberbandPlugin
 /*
 Modifies pitch of an audio without affecting duration
 */
+class PitchShift : public RubberbandPlugin
+
 {
 private:
   double _scaleFactor = 1.0;
 
+  // Allow pitch shifting by up to 6 octaves up or down:
+  static constexpr float MIN_SCALE_FACTOR = 0.015625;
+  static constexpr float MAX_SCALE_FACTOR = 64.0;
+
 public:
   void setScaleFactor(double scale) {
-    if (scale <= 0) {
-      throw std::range_error("Pitch scale must be a value greater than 0.0.");
+    if (scale < MIN_SCALE_FACTOR || scale > MAX_SCALE_FACTOR) {
```

Comment:
For c++20 nerds - looks like what you want is a [spaceship](https://en.cppreference.com/w/cpp/language/operator_comparison#Three-way_comparison)